### PR TITLE
Odyssey Widget: button to install Akismet

### LINKED ISSUE #76446: Odyssey Widget: Show an 'Install' button when Akismet is not intalled
Image

### PR DESCRIPTION
**Pull request by kangzj**
Related to #76446 

## Proposed Changes

Show `Install` button for the installation and activation of the Akismet plugin.

## Testing Instructions

* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Uninstall Akismet if installed
* Open `/wp-admin/index.php` on a site with Jetpack bleeding edge
* Ensure `Install` button shows in Akismet card
* Click the button
* Ensure Akismet is installed and activated
* If there's no key configured with Akismet
* Ensure a link is shown for user to manage their Akismet key

image

image

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/odyssey-stats/src/widget/modules.jsx [hunk 1]
```diff
@@ -94,9 +108,7 @@ function AkismetModule( { siteId } ) {
 			error={ akismetError?.message }
 			isLoading={ isAkismetLoading }
 			canManageModule={ canCurrentUser( siteId, 'manage_options' ) }
-			activateProduct={
-				canCurrentUser( siteId, 'manage_options' ) && activateProduct( 'anti-spam' )
```
Comment (kangzj) [inline apps/odyssey-stats/src/widget/modules.jsx:98]: permission checked in ModuleCard already

File: apps/odyssey-stats/src/widget/modules.jsx [hunk 2]
```diff
@@ -139,7 +151,7 @@ function ProtectModule( { siteId } ) {
 			error={ protectError?.message }
 			isLoading={ isProtectLoading }
 			canManageModule={ canCurrentUser( siteId, 'manage_options' ) }
-			activateProduct={ canCurrentUser( siteId, 'manage_options' ) && activateModule( 'protect' ) }
```
Comment (kangzj) [inline apps/odyssey-stats/src/widget/modules.jsx:142]: permission checked in ModuleCard already

File: apps/odyssey-stats/src/widget/modules.jsx [hunk 3]
```diff
@@ -19,10 +19,13 @@ function ModuleCard( {
 	isError = false,
 	canManageModule = false,
 	className = null,
-	info = null,
 } ) {
 	const translate = useTranslate();
 	const [ disabled, setDisabled ] = useState( false );
+	const activateAskismet = () => {
```
Comment (bluefuton) [inline apps/odyssey-stats/src/widget/modules.jsx]:
```suggestion
	const activateAkismet = () => {
```

File: apps/odyssey-stats/src/widget/modules.jsx [hunk 4]
```diff
@@ -42,22 +45,25 @@ function ModuleCard( {
 						<div className="stats-widget-module__info">
 							{ error === 'not_active' && (
 								<button
+									className="jetpack-emerald-button"
 									disabled={ disabled }
-									onClick={ () => {
-										setDisabled( true );
-										activateProduct().catch( () => setDisabled( false ) );
-									} }
+									onClick={ activateAskismet }
```
Comment (bluefuton) [inline apps/odyssey-stats/src/widget/modules.jsx]:
```suggestion
									onClick={ activateAkismet }
```

File: apps/odyssey-stats/src/widget/modules.jsx (hunk 3 above)
Comment (kangzj) [reply apps/odyssey-stats/src/widget/modules.jsx]: Good catch 👍 fixed in 705011a92f

File: apps/odyssey-stats/src/widget/modules.jsx (hunk 4 above)
Comment (kangzj) [reply apps/odyssey-stats/src/widget/modules.jsx]: Good catch 👍 fixed in 705011a92f

File: apps/odyssey-stats/src/widget/modules.jsx [hunk 5]
```diff
@@ -42,22 +46,30 @@ function ModuleCard( {
...
+									className="jetpack-emerald-button is-secondary-jetpack-emerald"
+									disabled={ disabled }
+									onClick={ onActivateProduct }
+								>
+									Install
+								</button>
+							) }
+							{ error === 'invalid_key' && (
+								<a href={ manageUrl } target="_self">
+									Manage Akismet Key
```
Comment (bluefuton) [inline apps/odyssey-stats/src/widget/modules.jsx]:
```suggestion
									Manage Akismet key
```

Nit: should be sentence case I think to match the heading above.

**Review by keoshi — APPROVED** [review]
Seems like a very sensible solution to me.

+1 on using sentence case as pointed out by @bluefuton!

**Pull request merged**